src/server/node_manager.c

```diff
@@ -7682,12 +7685,15 @@ free_resvNodes(resc_resv *presv)
 
 			DBPRT(("Freeing resvinfo on node %s from reservation %s\n",
 				pnode->nd_name, presv->ri_qs.ri_resvID))
-			if (prev == NULL)
+			if (prev == NULL) {
 				pnode->nd_resvp = rinfp->next;
-			else
+				free(rinfp);
+				rinfp = pnode->nd_resvp;
+			} else {
 				prev->next = rinfp->next;
-			free(rinfp);
-			break;
+				free(rinfp);
+				rinfp = prev->next;
+			}
```

Comment:
Since the free and the advance are in both sides of this if/else block, how about moving them after?

Reply:
Well in both cases free is common but the pointer is advanced using different structures. to make it common, I'll have to create a temporary pointer and advance rinfp using that pointer.
I'd like to retain it as it is unless you feel strongly about moving it.